pkg/templates/cel/template.go

```diff
@@ -0,0 +1,108 @@
+package cel
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/google/cel-go/cel"
+	"golang.stackrox.io/kube-linter/pkg/check"
+	"golang.stackrox.io/kube-linter/pkg/config"
+	"golang.stackrox.io/kube-linter/pkg/diagnostic"
+	"golang.stackrox.io/kube-linter/pkg/lintcontext"
+	"golang.stackrox.io/kube-linter/pkg/objectkinds"
+	"golang.stackrox.io/kube-linter/pkg/templates"
+	"golang.stackrox.io/kube-linter/pkg/templates/cel/internal/params"
+)
+
+const (
+	templateKey = "cel-expression"
+)
+
+func init() {
+	templates.Register(check.Template{
+		HumanName:   "CEL",
+		Key:         templateKey,
+		Description: "Flag objects with CEL expression",
+		SupportedObjectKinds: config.ObjectKindsDesc{
+			ObjectKinds: []string{objectkinds.Any},
+		},
+		Parameters:             params.ParamDescs,
+		ParseAndValidateParams: params.ParseAndValidate,
+		Instantiate: params.WrapInstantiateFunc(func(p params.Params) (check.Func, error) {
+			return func(ctx lintcontext.LintContext, object lintcontext.Object) []diagnostic.Diagnostic {
+				msg, err := evaluate(p.Check, object, ctx.Objects())
+				if err != nil {
+					return []diagnostic.Diagnostic{
+						{Message: fmt.Sprintf("error evaluating CEL check expression: %v", err)},
+					}
+				}
+				if msg != "" {
+					return []diagnostic.Diagnostic{
+						{Message: fmt.Sprintf("CEL check expression returned: %v", msg)},
+					}
+				}
+				return nil
+			}, nil
+		}),
+	})
+}
+
+func evaluate(check string, subject lintcontext.Object, objects []lintcontext.Object) (string, error) {
+	// Convert subject to map via JSON marshaling/unmarshaling for CEL compatibility
+	// We need to marshal the underlying K8sObject, not the lintcontext.Object
+	subjectMap, err := toMap(subject.K8sObject)
+	if err != nil {
+		return "", err
+	}
+
+	// Convert objects to maps via JSON marshaling/unmarshaling
+	objectsMaps := make([]map[string]any, len(objects))
+	for i, obj := range objects {
+		objMap, err := toMap(obj.K8sObject)
+		if err != nil {
+			return "", err
+		}
+		objectsMaps[i] = objMap
+	}
+
+	e, err := cel.NewEnv(
+		cel.Variable("subject", cel.MapType(cel.StringType, cel.AnyType)),
+		cel.Variable("objects", cel.ListType(cel.MapType(cel.StringType, cel.AnyType))),
+	)
+	if err != nil {
+		return "", err
```

Comment:
Wrap error? Here and everywhere in this function. Something like 

```suggestion
		return "", errors.Wrap(err, "setting up CEL environment")
```